### PR TITLE
Move sec_offset values out of the distinct data and take those values from the debug line tables

### PR DESCRIPTION
**Pull request by rastogishubham**
The DebugInfoSection blocks are not deduping as much as they should, this is because the sec_offset values for individual CUs change, which gets stored into the DistinctDataRef. This makes the DebugInfoSection Block also not dedupe, this is not an issue on a non-split case because these changing sec_offsets do not exist in the case where only one CU exists in the entire debug info.

To combat this, do not write a sec_offset at all, and when materializing, just grab it from the debug_line section

### CODE REVIEW

**Comment by rastogishubham** [comment]
@swift-ci please test

**Comment by rastogishubham** [comment]
@swift-ci please test

**Comment by rastogishubham** [comment]
@swift-ci please test

**Comment by rastogishubham** [comment]
@swift-ci please test

**Review by adrian-prantl — APPROVED** [review]

File: llvm/lib/MC/MCCASObjectV1.cpp [hunk 1]
```diff
@@ -1019,6 +1087,17 @@ static Error materializeCUDie(DWARFCompileUnit &DCU,
...
+      memcpy(&SectionContents[SectionOffset], &SecOffsetVals[SecOffsetIndex],
+             *FormSize);
+      // If the size of the SecOffsetsVals vector is 1 then we only have one
+      // line table, so its offset is the only one we should use for all
+      // sec_offsets.
+      if (SecOffsetVals.size() > 1)
+        SecOffsetIndex++;
+      else if (SecOffsetVals.size() == 0)
+        llvm_unreachable(
+            "SecOffset SmallVector should have a non-zero length!");
```
Comment (felipepiovezan) [inline llvm/lib/MC/MCCASObjectV1.cpp]: If we can express the same intent without control flow, it's easier to read:

```cpp
assert(SecOffsetVals.size() != 0 && "SecOffset SmallVector should have a non-zero length!"))
```

**Review by felipepiovezan — APPROVED** [review]
LGTM!
Left one minor suggestion.

Also, a small nit: in the future, I'd appreciate if renames were moved to their own NFC commit, as it is a lot easier to review that way. When renames are mixed with functional changes, reviewers need to go through a lot more code and figure out what is NFC and what is not.

File: llvm/lib/MC/MCCASObjectV1.cpp (hunk 1 above)
Comment (rastogishubham) [reply llvm/lib/MC/MCCASObjectV1.cpp]: That is fair!

**Pull request merged**